### ISSUE TITLE
Allow for Durable Entities Proxy to implement multiple interfaces

### ISSUE THREAD

**Issue by alvpickmans** [Azure/azure-functions-durable-extension#1276]
**Is your feature request related to a problem? Please describe.**
As per [documentation](https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-dotnet-entities#restrictions-on-entity-interfaces) regarding interfaces in durable entities, it specifies some constraints they must comply with. Although it doesn't explicitly say DEs classes can only implement a single interface, the code enforces it [here](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/EntityScheduler/TypedInvocationExtensions.cs#L104-L107) and [here](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/EntityScheduler/Proxy/EntityProxyFactory.cs#L91). As a simple example, the a DE class implementing the below `IPrimaryJob` below is not currently permitted:
```c#
       public interface IJob
        {
            void SetStartDate(DateTime date);
            void SetEndDate(DateTime date);
            Task<TimeSpan> GetDuration();
            void Delete();
        }

        public interface IPrimaryJob : IJob
        {
            void SetId(string Id);
            Task<string> GetId();
        }
```

**Describe the solution you'd like**
Being able to use multiple interfaces on DEs classes.

**Describe alternatives you've considered**
Duplicating interfaces to comply with current proxy limitations on DEs.

**Additional context**
I see the value of using global or more generic interfaces that children of these can extend to ensure and enforce a minimum common set of methods.

I have a PR ready that implements this with corresponding test.

**Comment by sebastianburckhardt** [comment]
I believe this is working as intended. The code snippets that throw an exception are executed only when using the "shorter version of the entity ID parameter", which is a feature for simplifying the common case where there is only one implementation per interface by omitting the entity name. This is documented [https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-dotnet-entities#shorter-option-for-specifying-the-target](here). 

If there is more than one implementation, one simply has to use the more verbose form of passing an entity id, which includes the entity name.

**Comment by alvpickmans** [comment]
The issue I encountered is not multiple classes implementing the same interface, but instead a single class implementing more than one interface. Taking the example below, the runtime throws the following exception:
```
System.Private.CoreLib: Exception while executing function: InitializeJob. System.Private.CoreLib: Method 'SetStartDate' in type 'IPrimaryJob_371d6bfd9a1243039db273fe245551a5' from assembly 'DynamicAssembly_31ef6567caf642b4979f0b68523f45ea, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
```

After digging, it is due to the `EntityProxyFactory` only getting methods for the given interface instead of all implemented ones. If this is the expected behavior cool, but I feel that it reduces the options when designing durable entities using interfaces. PR #1277 addresses this without breaking changes

```c#
namespace Azure.Durable.MultipleInterfaces
{
    public interface IJob
    {
        void SetStartDate(DateTime date);
        void SetEndDate(DateTime date);
    }

    public interface IPrimaryJob : IJob
    {
        void SetId(string Id);
    }

    public class DurablePrimaryJob : IPrimaryJob
    {
        public string Id { get; private set; }
        public DateTime StartDate { get; private set; }
        public DateTime EndDate { get; private set; }
        public void SetId(string Id) => this.Id = Id;
        public void SetEndDate(DateTime date) => this.EndDate = date;
        public void SetStartDate(DateTime date) => this.StartDate = date;

        [FunctionName(nameof(DurablePrimaryJob))]
        public static Task Run(
        [EntityTrigger] IDurableEntityContext context)
            => context.DispatchAsync<DurablePrimaryJob>();
    }

    public static class DurableEntitiesFunctions
    {
        [FunctionName(nameof(RunOrchestrator))]
        public static async Task RunOrchestrator(
            [OrchestrationTrigger] IDurableOrchestrationContext context)
        {
            await context.CallActivityAsync(nameof(InitializeJob), (context.InstanceId, context.CurrentUtcDateTime));
        }

        [FunctionName(nameof(InitializeJob))]
        public static async Task InitializeJob(
            [ActivityTrigger] (string orchestratorId, DateTime start) payload,
            [DurableClient] IDurableClient client,
            ILogger log)
        {
            EntityId id = new EntityId(nameof(DurablePrimaryJob), payload.orchestratorId);
            string jobId = Guid.NewGuid().ToString();
            await client.SignalEntityAsync<IPrimaryJob>(id, x => x.SetId(jobId));
            await client.SignalEntityAsync<IJob>(id, x => x.SetStartDate(payload.start));
        }

        [FunctionName(nameof(HttpStart))]
        public static async Task<HttpResponseMessage> HttpStart(
            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = "primaryjob")] HttpRequestMessage req,
            [DurableClient] IDurableOrchestrationClient starter,
            ILogger log)
        {
            // Function input comes from the request content.
            string instanceId = await starter.StartNewAsync(nameof(RunOrchestrator));

            log.LogInformation($"Started orchestration with ID = '{instanceId}'.");
            return starter.CreateCheckStatusResponse(req, instanceId);
        }
    }
}
```

**Comment by sebastianburckhardt** [comment]
Ok, sorry for misunderstanding! I will have a look at the PR.

**Comment by ConnorMcMahon** [comment]
Addressed in #1277
